python/pyspark/sql/dataframe.py

```diff
@@ -1272,16 +1272,18 @@ def replace(self, to_replace, value, subset=None):
         """Returns a new :class:`DataFrame` replacing a value with another value.
         :func:`DataFrame.replace` and :func:`DataFrameNaFunctions.replace` are
         aliases of each other.
+        Values `to_replace` and `value` should be homogeneous. Mixed string and numeric
```

Comment:
Oh, you mean when we use higher precision type with lower precision column. Not really an issue with `replace`, don't you think? Maybe something like _`value` will be truncated when used with lower precision column_? I doubt we can make it precise and comprehensible at the same time and the main goal should be to signal possible issues while keeping things relatively clear. After all there are four different signatures to cover here.

I've missed boolean case. I keep forgetting that `True` is instance of  `int`.  Added to the other PR, with tests for mixed numeric types.